lib/node-pre-gyp.js

```diff
@@ -9,6 +9,13 @@ module.exports = exports;
  * Module dependencies.
  */
 
+// load mocking control function for accessing s3 via https. the function is a noop always returning
+// false if not mocking.
+exports.mockS3Http = require('./util/s3_setup').get_mockS3Http();
+exports.mockS3Http('on');
+const mocking = exports.mockS3Http('get');
```

Comment:
Could you add a little more detail on why this is needed? My question is coming from the deeper question of whether there is any way we could avoid having this code in `lib/`. If not I don't think this is a blocker, but want to understand more why it is needed.

Reply:
my original goal (in the original PR) was to not modify any of the code in lib (or as little as possible). but to do so meant modifying the script command in package.json which led to more problems (different env var settings for windows/linux, changes in what program was actually run, etc. - that stuff is still in the other PR if you want to compare).

i originally put this here because i am not sure whether `nock` patches the http/https modules after they have been loaded or only when they are loaded. if only when they are loaded, i wanted to load nock as early as possible. i did not get around to testing that so i don't know.

but one way or the other the code has to go into `lib/`. the reason is that it is `lib/install` that needs to use the mocked https endpoint, not the test module. while `test/fetch.test.js` requires `lib/install.js` and invokes the function directly, the `test/build.test.js` tests execute `bin/node-pre-gyp` via a subprocess and there is no good way that i came up with (see original PR for **not-so-good** way) to have the mock code in the same node instance in that case.

Reply:
and, if not in `lib/node-pre-gyp.js` it would have to go into `lib/install.js` which seemed worse.